site/gatsby-site/src/contexts/userContext/UserContextProvider.js

```diff
@@ -44,27 +46,60 @@ export const UserContextProvider = ({ children }) => {
 
   const [user, setUser] = useState(realmApp.currentUser);
 
+  const { t } = useTranslation();
+
+  const addToast = useToastContext();
+
   const logout = async () => {
     await realmApp.currentUser.logOut();
 
     //TODO: functions need at least an anon user to work
     await login();
   };
 
-  const login = async ({ email = null, password = null } = {}) => {
-    let credentials = null;
+  const login = async ({
+    email = null,
+    password = null,
+    provider = null,
+    loginRedirectUri = null,
+  } = {}) => {
+    try {
+      let credentials = null;
+
+      if (email && password) {
+        credentials = Realm.Credentials.emailPassword(email, password);
+      } else if (provider === 'facebook' && loginRedirectUri) {
+        credentials = Realm.Credentials.facebook(loginRedirectUri);
+      } else if (provider === 'google' && loginRedirectUri) {
+        credentials = Realm.Credentials.google(loginRedirectUri);
+      } else {
+        credentials = Realm.Credentials.anonymous();
+      }
 
-    if (email && password) {
-      credentials = Realm.Credentials.emailPassword(email, password);
-    } else {
-      credentials = Realm.Credentials.anonymous();
+      const user = await realmApp.logIn(credentials);
+
+      if (user.id === realmApp.currentUser.id) {
+        setUser(user);
+      }
+    } catch (e) {
+      console.error(e);
```

Comment:
I think we can delete this and later implement: https://github.com/responsible-ai-collaborative/aiid/issues/586